src/map/script.c

```diff
@@ -5733,67 +5781,22 @@ BUILDIN(callfunc)
 	ri->defsp        = st->stack->defsp;        // default stack pointer
 	script->push_retinfo(st->stack, ri, ref);
 
-	st->pos = 0;
-	st->script = scr;
-	st->stack->defsp = st->stack->sp;
-	st->state = GOTO;
-	st->stack->scope.vars = i64db_alloc(DB_OPT_RELEASE_DATA);
-	st->stack->scope.arrays = idb_alloc(DB_OPT_BASE);
-
-	if( !st->script->local.vars )
-		st->script->local.vars = i64db_alloc(DB_OPT_RELEASE_DATA);
-
-	return true;
-}
-/*==========================================
- * subroutine call
- *------------------------------------------*/
-BUILDIN(callsub)
-{
-	int i,j;
-	struct script_retinfo* ri;
-	int pos = script_getnum(st,2);
-	struct reg_db *ref = NULL;
-
-	if( !data_islabel(script_getdata(st,2)) && !data_isfunclabel(script_getdata(st,2)) )
+	if (pos != -1 && !scr)
+		st->pos = pos;
+	else
 	{
-		ShowError("script:callsub: argument is not a label\n");
-		script->reportdata(script_getdata(st,2));
-		st->state = END;
-		return false;
+		st->pos = (scr) ? 0 : nd->u.scr.label_list[k].pos;
```

Comment:
Not really see some parts of the code:
```c
ARR_FIND(0, nd->u.scr.label_list_num, k, (strcmp(nd->u.scr.label_list[k].name, labelname) == 0));
```
ARR_FIND = Finds an entry in an array and increment 'param _var' in this case 'k'.

And in this part:
```c
st->pos = (scr) ? 0 : nd->u.scr.label_list[k].pos;
```

And 'k' will be used if 'scr' is 'NULL' (Not found function) and 'ARR_FIND' found the given label.